docs/AutomationProtocols.md

```diff
@@ -0,0 +1,98 @@
+
+### Disadvantages
+
+- Not designed for in depth browser analysis (e.g. tracing or intercepting network events)
+- limited set of automation capabilities (e.g. no support to throttle CPU or network)
+
+## DevTools Protocol
+
+The DevTools interface is a native browser interface that is usually being used to debug the browser from a remote application (e.g. [Chrome DevTools](https://developers.google.com/web/tools/chrome-devtools/)). Next to its capabilities to inspect the browser in nearly all possible forms it can also be used to control it. While evey browser used to have its own internal DevTools interface that was not much expose to the user, more and more browser now adapt the [Chrome DevTools Protocol](https://chromedevtools.github.io/devtools-protocol/) that is used to either debug a web application using Chrome DevTools or control Chrome using tools like [Puppeteer](https://pptr.dev/). The communication happens without any proxy, directly to the browser using WebSockets:
+
+![DevTools Setup](/img/devtools.png)
+
+WebdriverIO allows you to use this automation technology as an alternative for WebDriver if you have special requirements to automate the browser. With the [`devtools`](https://www.npmjs.com/package/devtools) NPM package you can use the same commands that WebDriver provides which then can be used by WebdriverIO and the WDIO testrunner to run its useful commands on top of that protocol. It uses Puppeteer to under the hood and allows you to run a sequence of commands with Puppeteer if need to. To use DevTools as automation protocol you have to install the NPM package as follows:
+
+```sh
+$ npm i --save-dev devtools
+```
+
+In your WebdriverIO options you then just need to switch the `automationProtocol` flag to `devtools`, e.g. in your `wdio.conf.js`:
+
+```js
+// wdio.conf.js
+exports.config = {
+    // ...
+    automationProtocol: 'devtools'
+    // ...
+}
+```
+
+Then you can run a test like this:
+
+```js
+// example using WDIO testrunner
+describe('my test', () => {
+    it('can use Puppeteer as automation fallback', () => {
+        // WebDriver command
+        browser.url('https://webdriver.io')
+
+        // switch to Puppeteer to intercept requests
+        browser.call(async () => {
+            const puppeteerBrowser = browser.getPuppeteer()
+            const page = (await puppeteerBrowser.pages())[0]
+            await page.setRequestInterception(true)
+            page.on('request', interceptedRequest => {
+                if (interceptedRequest.url().endsWith('webdriverio.png')) {
+                    return interceptedRequest.continue({
+                        url: 'https://webdriver.io/img/puppeteer.png'
+                    })
+                }
+
+                interceptedRequest.continue()
+            })
+        })
+
+        // continue with WebDriver commands
+        browser.url('https://webdriver.io')
+
+        /**
+         * WebdriverIO logo is no replaced with the Puppeteer logo
```

Comment:
this comment can be removed it seems

Reply:
I used it to describe the result of the example code